### ISSUE TITLE
Spider logs more than actual found URIs.

### ISSUE THREAD

**Issue by jeremychoi** [zaproxy/zaproxy#7737]
### Describe the bug

Job spider is reporting always 3 or more even when there is no URLs that can be found, like the following:

"Job spider found 3 URLs" 

The logs above came from Automation Framework's messages but the issue seems to exist in the spider addon itself (e.g. https://github.com/zaproxy/zap-extensions/blob/420d2e9d24c44f6a54089c54ad432531cf336b9c/addOns/spider/src/main/java/org/zaproxy/addon/spider/SpiderController.java#L193)

I assume the URLs are including the user-suppiled URL (e.g. a default target URL) plus robots.txt and sitemap.xml which Spider sends automatically. 

The root cause might be, increasing the counts regardless of the return status (e.g. 404).

### Steps to reproduce the behavior

It can be reproduced using any web apps. Just a simple example can be:
1. create an empty dir
2. python3 -m http.server 5000
3. run spider with automation framework 

### Expected behavior

It should report counts of existing pages only.

### Software versions

ZAP 2.12.0

### Would you like to help fix this issue?

- [X] Yes

**Comment by thc202** [comment]
That's the expected behaviour, status code is not relevant for the count of URLs found. Though ignoring the seeds would probably be more accurate since those are technically not found while spidering.

**Comment by jeremychoi** [comment]
I see. You're right. If a URL is found and returned with 404, that should be counted too.  And, I agree it will make it accurate to ignore the seeds

**Comment by jeremychoi** [comment]
thanks for assigning. I will be able to work in April.

**Comment by kingthorin** [comment]
@jeremychoi do you still plan to tackle this?

**Comment by jeremychoi** [comment]
@kingthorin  yes. sorry for the delay. I couldn't find time to work on it. I'll do this Q. However, if there's someone else who wants to fix it, it's okay for this to be reassigned.

**Comment by kingthorin** [comment]
No problem and no rush. Life gets busy.

I’m working on something else with the Spider but might tackle this in a few weeks if it’s still kicking around.
